PrestaShop/vatnumber
# Refactoring validation through webservice

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Refactoring to handle European Commission web service easier
| Type?         | refacto
| BC breaks?    | yes, now require PHP 7.1 so available start PrestaShop 1.7.7.0
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | follow instructions below

To be released with 1.7.7.0 ?

## Changes
* Introduce dependency https://github.com/DragonBe/vies (MIT license)
* Require PHP 7.1 so PrestaShop 1.7.7.0 is now minimum supported version
* Validate a VAT number control sum without using web service
* Validate a VAT number through European Commission web service (optional)
* Adds logs when web service is unavailable
* Adds a ModuleAdminController to manage configuration in multistore more easier thanks to legacy HelperOptions
* Adds devtools and GitHub Actions
* Remove unused ajax.php file (Used only before PrestaShop 1.7)

A new logo would be awesome 🙏

## How to test

PrestaShop 1.7.7.0 minimum

In BO > module configuration page
If SOAP extension is not available "Validate through web service" is disabled

In FO > My Account > Addresses > Add or Edit an Address
Field Company : PrestaShop
Field VAT number : FR66497916635

If syntax of VAT number is invalid > Error "This VAT number is invalid."
If syntax of VAT number is valid and validation through web service enabled but VAT number is invalid > Error "This VAT number is invalid."

If validation through web service is enabled but web service is not available a new log entry should be visible in BO > Advanced Parameters > Logs


🚧  WIP 🚧 
Call VIES only if customer country is inside EU like https://github.com/PrestaShop/vatnumber/pull/14
